net/quic/packet.c

```diff
@@ -509,6 +510,7 @@ static struct sk_buff *quic_packet_create(struct sock *sk, struct quic_packet_in
 			continue;
 		}
 		quic_outq_rtx_tail(sk, fskb);
+		QUIC_SND_CB(fskb)->level = packet->level;
```

Comment:
Never change QUIC_SND_CB(fskb)->level once fskb was created.  For 0-RTT DATA, it will be ACKed by 1-RTT packet, for which it needs to match both QUIC_SND_CB(fskb)->level and packet_number in quic_outq_retransmit_check().

In fact, here in quic_packet_create() `packet->level` is always QUIC_CRYPTO_APP, in quic_packet_handshake_create(), `level` is always QUIC_CRYPTO_APP, the same as QUIC_SND_CB(fskb)->level for 0-RTT DATA.